packages/twenty-front/src/modules/activities/components/ActivityBodyEditor.tsx

```diff
@@ -145,9 +164,76 @@ export const ActivityBodyEditor = ({
     }
   };
 
+  useListenClickOutside({
+    refs: [editorRef],
+    callback: (event) => {
+      if (
+        (event.target as HTMLDivElement)?.id === containerId &&
+        currentHotkeyScope?.scope === RightDrawerHotkeyScope.RightDrawer
```

Comment:
You could even use a dedicated recoil state

Reply:
Hi @lucasbordeau can you please check this commit https://github.com/twentyhq/twenty/pull/3568/commits/e9344eb9ebe915ac47f515226077e09f38fcba18

is this what expected using state??

Reply:
ok will update

Reply:
Except for record inline cell, rest of the field props are fine? @lucasbordeau

Reply:
I can't see your changes on this PR, have you pushed them already ?

Reply:
sorry i was referring to the commit that I shared here https://github.com/twentyhq/twenty/pull/3568#discussion_r1473384144

Reply:
Additionally, keyboard input hot key scope now checks for state instead of rightdrawer scope, please have a look

Am not sure of the following point ` we should use something like useRegisterInputEvents and not add new props to RecordInlineCell`

Reply:
Ok sorry for the delay I have been working on a large refactor.

I merged main into your PR manually because I had to work on activity editor too.

I added some comments to guide you through the completion, mainly it's about not drilling down onBlur and onFocus but directly setting the recoil state from the child components : ActivityTitle, ActivityBodyEditor and ActivityComments.

Recoil states can be called from anywhere and we want to avoid having everything in the parent component.

Reply:
You can remove onBlur and onFocus also on ActivityTargetsInlineCell and just set the recoil state here too in a `handleInlineCellClick`